### PR TITLE
TS-5024: Limits the gzip to 200 OK responses

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1133/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1027/ for details.

File: plugins/gzip/gzip.cc [hunk 1]
```diff
@@ -480,8 +480,8 @@ gzip_transformable(TSHttpTxn txnp, bool server, HostConfiguration *host_configur
 
   resp_status = TSHttpHdrStatusGet(bufp, hdr_loc);
 
-  // conservatively pick some statusses to compress
-  if (!(resp_status == 200 || resp_status == 404 || resp_status == 500)) {
+  // conservatively pick some statusses to compress. ToDo: Make this configurable?
+  if (resp_status != 200) {
```
Comment (oschaaf) [inline plugins/gzip/gzip.cc]: This seems like a good change to me, it's the probably safer default -- but is it possible that we see `200/OK` two times  here when the HttpSM is restarted?

Comment (zwoop) [inline plugins/gzip/gzip.cc]: @oschaaf  It is *possible*, but unlikely. The immediate issues we're having is that gzip works very poorly together with the escalate plugin, but you wouldn't escalate on a 200 OK. Long term, @meeramn is going to add a new plugin API, that enforces uniqueness on the continuations added to a hook.

Comment (oschaaf) [inline plugins/gzip/gzip.cc]: I see. LGTM, no more questions

**Pull request closed without merging**